api/lcp/server.py

```diff
@@ -0,0 +1,351 @@
+
+
+class LCPServerConfiguration(ConfigurationBucket):
+    """Contains LCP License Server's settings"""
+
+    DEFAULT_PAGE_SIZE = 100
+    DEFAULT_PASSPHRASE_HINT = 'If you do not remember your passphrase, please contact your administrator'
+    DEFAULT_ENCRYPTION_ALGORITHM = HashingAlgorithm.SHA256.value
+    DEFAULT_MAX_PRINTABLE_PAGES = 10
+    DEFAULT_MAX_COPIABLE_PAGES = 2048
+
+    lcpserver_url = ConfigurationMetadata(
+        key='lcpserver_url',
+        label=_('LCP License Server\'s URL'),
+        description=_('URL of the LCP License Server'),
+        type=ConfigurationAttributeType.TEXT,
+        required=True
+    )
+
+    lcpserver_user = ConfigurationMetadata(
+        key='lcpserver_user',
+        label=_('LCP License Server\'s user'),
+        description=_('Name of the user used to connect to the LCP License Server'),
+        type=ConfigurationAttributeType.TEXT,
+        required=True
+    )
+
+    lcpserver_password = ConfigurationMetadata(
+        key='lcpserver_password',
+        label=_('LCP License Server\'s password'),
+        description=_('Password of the user used to connect to the LCP License Server'),
+        type=ConfigurationAttributeType.TEXT,
+        required=True
+    )
+
+    lcpserver_input_directory = ConfigurationMetadata(
+        key='lcpserver_input_directory',
+        label=_('LCP License Server\'s input directory'),
+        description=_(
+            'Full path to the directory containing encrypted books. '
+            'This directory should be the same as lcpencrypt\'s output directory'
+        ),
+        type=ConfigurationAttributeType.TEXT,
+        required=True
+    )
+
+    lcpserver_page_size = ConfigurationMetadata(
+        key='lcpserver_page_size',
+        label=_('LCP License Server\'s page size'),
+        description=_('Number of licences returned by the server'),
+        type=ConfigurationAttributeType.NUMBER,
+        required=False,
+        default=DEFAULT_PAGE_SIZE
+    )
+
+    provider_name = ConfigurationMetadata(
+        key='provider_name',
+        label=_('LCP service provider\'s identifier'),
+        description=_(
+            'URI that identifies the provider in an unambiguous way'
+        ),
+        type=ConfigurationAttributeType.TEXT,
+        required=True
+    )
+
+    passphrase_hint = ConfigurationMetadata(
+        key='passphrase_hint',
+        label=_('Passphrase hint'),
+        description=_('Hint proposed to the user for selecting its passphrase'),
+        type=ConfigurationAttributeType.TEXT,
+        required=False,
+        default=DEFAULT_PASSPHRASE_HINT
+    )
+
+    encryption_algorithm = ConfigurationMetadata(
+        key='encryption_algorithm',
+        label=_('Passphrase encryption algorithm'),
+        description=_('Algorithm used for encrypting the passphrase'),
+        type=ConfigurationAttributeType.TEXT,
+        required=False,
+        default=DEFAULT_ENCRYPTION_ALGORITHM
```

Comment:
How many different encryption algorithms are there? This should be a selection with the valid values prepopulated, not a freeform text field.

Reply:
I changed type to `ConfigurationAttributeType.SELECT`